File: 프로그래머스/lv2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    int r = left / n, c = left % n;
    int tmp = r + ((c > r) ? c - r : 0) + 1;
    for (long long i = left; i <= right; i++) {
        answer.push_back(tmp);
        if (++c >= n) {
            c = 0;
            r++;
            tmp = r + 1;
        }
        if (c > r) tmp++;
    }
    return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.*;

class Solution {
    public int[] solution(int n, long left, long right) {
        long range = right - left + 1;
        int[] res = new int[(int)range];
        for (int i = 0; i < range; i++) {
            int row = (int)((left + i) / n);
            int col = (int)((left + i) % n);
            res[i] = Math.max(row, col) + 1;
        }
        return res;
    }
}